src/nodejs-common/operation.ts

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2022 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Metadata,
+  MetadataCallback,
+  ServiceObject,
+  ServiceObjectConfig,
+} from './service-object';
+import {ApiError} from './util';
+import {promisify} from 'util';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class Operation<T = any> extends ServiceObject<T> {
```

Comment:
This generic can be "tightened" in a future PR - we can pass an expected interface here